### PR TITLE
Add gitHead in subdirectories too

### LINKED ISSUE #66: gitHead not populated if we operate out of a sub directory
_Edit:_ [see my later comment for a better description of the issue](https://github.com/npm/read-package-json/issues/66#issuecomment-222036879)

Could we use something like

```
git rev-parse --show-toplevel
```

to find the directory? Otherwise, when publishing out of subdirectories, the git root isn't found.

Also: how would this be implemented? Running git directly?

### PR DESCRIPTION
**Pull request by felixfbecker**
Fixes #66

### CODE REVIEW

**Comment by felixfbecker** [comment]
CI failure is because latest npm is installed on Node 4

**Comment by felixfbecker** [comment]
@zkat @iarna could one of you give this a review? :)

**Comment by felixfbecker** [comment]
Friendly ping? This would really help a bunch of tools in the npm ecosystem

File: read-json.js [hunk 1]
```diff
@@ -329,12 +329,18 @@ function githead (file, data, cb) {
...
   fs.readFile(head, 'utf8', function (er, head) {
-    if (er) return cb(null, data)
-    githead_(file, data, dir, head, cb)
+    if (er) {
+      var parent = path.dirname(dir)
+      if (parent === dir) {
+        return cb(null, data)
+      }
+      return githead(dir, data, cb)
+    }
```
Comment (zkat) [inline read-json.js]: Shouldn't this be `githead_`?

Comment (felixfbecker) [reply read-json.js]: I don't think so - `githead()` reads the `.git/HEAD` file (if found) and passes the content to `githead_()`. So to traverse up the tree and find the `.git/HEAD` file, we need to call `githead()` recursively.
`githead_()` is what then handles the contents of the `.git/HEAD` file by reading the ref it points to etc.

It's easy to get confused though with the naming of the functions and no comments...

**Review by lastthyme — APPROVED** [review]

**Review by iarna — APPROVED** [review]
This looks good to me (but note: I'm not a committer here now) and per the RFC meeting either it will get merged or something that does the same thing will be. Personally I'd like to see this get merged as is to get the feature out there and in the ecosystem as early as possible.

**Comment by wraithgar** [comment]
Thanks for your patience here. I'm getting this repo's CI up to date and then we can rebase, relint, retest, and ship it.

Note that `gitHead.js` was renamed because case sensitive filenames are something we are trying to avoid.

**Comment by wraithgar** [comment]
https://github.com/npm/read-package-json/pull/101

**Pull request closed without merging**